### PR TITLE
feat: spec width in bptree

### LINKED ISSUE #152: Further improve value storage
Each b+ tree corresponds to a single type, so in the case of fixed width types (basically everything except a string), we don't actually need to store the value's length and instead can figure it out from the type of the field. Note, we still need the data pointer as a disambiguator.

In essence, we should pass some information into the b+ tree node that indicates that it's either fixed width or variable width, and if it's fixed width, how wide it should be. Widths are:
* 1 byte for boolean
* 8 bytes for int64 and float64
* 0 bytes for null

### PR DESCRIPTION
**Pull request by friendlymatthew**
Closes #152 

`DetermineWidth(ft FieldType)` determines the width, where `0` represents variable width and `1 + x` represents the fixed width `x`.

We drill this through the `bptree` lifecycle, updating how we write and read from disk.

We store the width in `IndexMeta` since we currently following a singleton relationship between a `MetaPage` and `Bptree`. 

Read accordingly to typescript.

- [x] fixed tests
- [x] behaves accordingly on demo

### CODE REVIEW

File: src/tests/mock_binaries/green_tripdata_2023-01.index
Comment (kevmo314) [inline src/tests/mock_binaries/green_tripdata_2023-01.index:1]: Does this have to be checked in?

**Review by kevmo314 — APPROVED** [review]

File: src/tests/mock_binaries/green_tripdata_2023-01.index
Comment (friendlymatthew) [reply src/tests/mock_binaries/green_tripdata_2023-01.index:1]: We use this file in `index-file.test` to read the file meta, and make sure we can traverse and collect the pages correctly

Comment (kevmo314) [reply src/tests/mock_binaries/green_tripdata_2023-01.index:1]: Can we generate it instead? It seems like an integration test which is good to have but having fixtures will be a maintenance burden later if we update the schema.

Comment (friendlymatthew) [reply src/tests/mock_binaries/green_tripdata_2023-01.index:1]: Yeah that makes sense -- I'll update accordingly

**Pull request merged**